web/router.go

```diff
@@ -84,17 +84,19 @@ func assembleRoutes() {
 	ramlapi.Build(api, routerFunc)
 }
 
-func routerFunc(data map[string]string) {
+func routerFunc(ep *ramlapi.Endpoint) {
 	route := router.
-		Methods(data["verb"]).
-		Path(data["path"]).
-		Handler(RouteMap[data["handler"]])
+		Methods(ep.Verb).
+		Path(ep.Path).
+		Handler(RouteMap[ep.Handler])
 
-	if data["query"] != "" {
-		if data["query_pattern"] != "" {
-			route.Queries(data["query"], fmt.Sprintf("{%s:%s}", data["query"], data["query_pattern"]))
-		} else {
-			route.Queries(data["query"], "")
+	if len(ep.QueryParameters) != 0 {
+		for _, query := range ep.QueryParameters {
```

Comment:
Splitting hairs here, but I think ```param``` is a better name than ```query```. Query could imply the entire query string.